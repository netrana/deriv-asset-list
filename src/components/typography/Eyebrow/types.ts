export interface Props {
  className?: string;
  isMuted?: boolean;
}
