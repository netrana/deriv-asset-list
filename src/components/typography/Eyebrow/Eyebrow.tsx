import React, { FC } from 'react';
import { Typography } from 'antd';
import classNames from 'classnames';

import { Props } from './types';

import styles from './eyebrow.module.scss';

const Eyebrow: FC<Props> = (props) => {
  const { children, isMuted = false, className = '' } = props;

  return (
    <Typography.Text
      className={classNames({
        [styles.eyebrow]: true,
        [styles.isMuted]: isMuted,
        [className]: Boolean(className),
      })}
    >
      {children}
    </Typography.Text>
  );
};

export default Eyebrow;
