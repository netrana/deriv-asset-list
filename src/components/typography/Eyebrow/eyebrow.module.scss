@import 'styles/globals.scss';

.eyebrow {
  font-weight: $fontWeightMedium;
  font-size: 16px;
  line-height: 16px;
  letter-spacing: 0.08em;
  margin: 0 !important;

  &.isMuted {
    opacity: 0.8;
  }
}
