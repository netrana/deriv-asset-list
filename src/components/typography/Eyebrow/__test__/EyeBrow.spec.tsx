import React from "react";
import render from "react-test-renderer";

import Eyebrow from "../Eyebrow";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  FC: jest.fn(),
}));

jest.mock("antd", () => ({
  ...jest.requireActual("antd"),
  Typography: {
    ...jest.requireActual("antd").Typography,
    Text: "Text",
  },
}));

describe("Eyebrow component test", () => {
  describe("snapshot test", () => {
    it("should match snapshot with default props", () => {
      const tree = render.create(<Eyebrow>Test Text</Eyebrow>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should match snapshot with isMuted true", () => {
      const tree = render.create(<Eyebrow isMuted>Test Text</Eyebrow>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
