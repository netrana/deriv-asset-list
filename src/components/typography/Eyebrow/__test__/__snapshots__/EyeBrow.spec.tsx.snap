// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eyebrow component test snapshot test should match snapshot with default props 1`] = `
<Text
  className="eyebrow"
>
  Test Text
</Text>
`;

exports[`Eyebrow component test snapshot test should match snapshot with isMuted true 1`] = `
<Text
  className="eyebrow isMuted"
>
  Test Text
</Text>
`;
